docs/getting_started/basic_usage/basic_usage.rst

```diff
@@ -27,58 +27,26 @@ as well as some training parameters.
 Some common config parameters:
 
 task
-    The task, such as ``train`` or ``forward``.
+    The task, such as ``train``, ``forward`` or ``search``.
 
 device
     E.g. ``gpu`` or ``cpu``. Can also be ``gpu0,gpu1`` for multi-GPU training.
 
 use_tensorflow
     If you set this to ``True``, TensorFlow will be used.
 
-train / dev
+train / dev / eval
     The datasets. This can be a filename to a hdf-file.
     Or it can be a dict with an entry ``class`` where you can choose a from a variety
     of other dataset implementations, including many synthetic generated data.
+    ``train`` and ``dev`` are used during training, while ``eval`` is usually used to define the dataset for the
+    ``forward`` or ``search`` task.
```

Comment:
Not quite correct. In general, I would not set `eval` inside the config. It would also be used during training then (in contrast to what you write), and not sure if this is what people expect. For the `"search"` task, I would esp recommend to use the `"search_data"` option instead.

Reply:
What about forward mode? For me this was never an issue, because i need if cases to distinguish between training and testing anyways, and eval would never be set during training. I never had any problems with using eval, neither for forward nor for search, and i did not even know about the search_data option.

But anyways, this exactly the reason why i started writing the documentation, as many things seem to be unclear for me.

Reply:
Well, I intentionally left that case out of the documentation because it was not quite well-defined and the behavior could potentially change. (I would change it, but I'm not sure if that breaks someones setup...)

Reply:
From what I see in the code, following happens:
 * during `task='train'`, it will call `eval_model()` which evaluates on `dev` and `eval`, as `get_eval_dataset()` returns both. 
 * the same holds for `task='eval'`, unless `output_per_seq_file` is specified, then it asserts that only either (dev or eval, but not both) are set.
 * `task='forward'` only uses `eval` , nothing else is used
 * `task='search` uses `eval` as default,  but it can use `train` or `dev` if `seach_data` is set to `'train'` or `'dev'`. If `search_data` is not a string, it will interpret this as dataset.

You are definitely right, that this should be changed. So what of this should be part of the documentation?

Reply:
I'm not sure about changing the logic too much, as I don't know who actually depends on this behavior. But also, that is why I would not document `eval`, because the behavior is somewhat inconsistent, and new users should better just not use it at all. For the `search` task, people could just use `seach_data` (that's what I recommend anyway). We could independently document that for the search task. The forward task is anyway not documented in detail yet (and probably anyway rarely used; there are other things which are probably more important). And if needed, we could introduce a new key `forward_data` or so for that case.